packages/client/src/features/tools/viewport/movement-key-tool.ts

```diff
@@ -0,0 +1,177 @@
+/********************************************************************************
+ * Copyright (c) 2023 Business Informatics Group (TU Wien) and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { Action, ChangeBoundsOperation, Point, SetViewportAction, Viewport } from '@eclipse-glsp/protocol';
+import { inject, injectable, optional } from 'inversify';
+import {
+    KeyListener,
+    KeyTool,
+    SModelElement,
+    findParentByFeature,
+    isViewport,
+    isSelectable,
+    isBoundsAware,
+    SModelRoot,
+    ISnapper
+} from 'sprotty';
+import { matchesKeystroke } from 'sprotty/lib/utils/keyboard';
+import { SelectableBoundsAware } from '../../../utils/smodel-util';
+import { GLSPTool } from '../../../base/tool-manager/glsp-tool-manager';
+import { TYPES } from '../../../base/types';
+import { GridSnapper } from '../../change-bounds/snap';
+
+/**
+ * Moves viewport when its focused and arrow keys are hit.
+ */
+@injectable()
+export class MovementKeyTool implements GLSPTool {
+    static ID = 'glsp.movement-key-tool';
+
+    isEditTool = true;
+
+    protected readonly movementKeyListener = new MoveKeyListener(this);
+
+    @inject(KeyTool) protected readonly keytool: KeyTool;
+    @inject(TYPES.ISnapper) @optional() readonly snapper?: ISnapper;
+    get id(): string {
+        return MovementKeyTool.ID;
+    }
+
+    enable(): void {
+        this.keytool.register(this.movementKeyListener);
+    }
+
+    disable(): void {
+        this.keytool.deregister(this.movementKeyListener);
+    }
+}
+
+export class MoveKeyListener extends KeyListener {
+    protected grid = { x: 20, y: 20 };
```

Comment:
I'm not a big fan of magic numbers in general. But in this case it is used if there is no available `GridSnapper`, right? Could you please add some documentation here and for example extract the default values into static members, something like:

```suggestion
    static readonly defaultMoveX = 20;
    static readonly defaultMoveY = 20;

    protected grid = { x: MoveKeyListener.defaultMoveX, y: MoveKeyListener.defaultMoveY };
```